src/jit/valuenum.cpp

```diff
@@ -1188,7 +1265,7 @@ ValueNum ValueNumStore::VNForMapSelectWork(ValueNumKind vnk, var_types typ, Valu
     VNDefFunc2Arg fstruct(VNF_MapSelect, arg0VN, arg1VN);
     if (GetVNFunc2Map()->Lookup(fstruct, &res))
     {
-        return res;
+        goto Return;
```

Comment:
Consider:
Instead of using goto Return and goto TailCall, it is possible to write this using a loop and use break and continue.